src/librustc_trans/back/link.rs

```diff
@@ -670,12 +670,6 @@ fn link_natively(sess: &Session,
         if !out.contains(msg) {
             break
         }
-
```

Comment:
Was this intentionally removed?

Reply:
@Mark-Simulacrum Yes, it is explained in d993ffe.

Reply:
Hm, I wonder if we can instead send the warning to stdout or something -- non ideal, I agree, but I feel like saying nothing, always, isn't a good idea. At least maybe adding a call to `info!()`? @alexcrichton probably has some thoughts here too

Reply:
@Mark-Simulacrum Since the stdout is also tested in the UI test, probably not possible either. Printing to the log crate should be good since logs are not enabled by UI test, will do that after CI is completed.

Reply:
Yeah I'm sort of ok with whatever. I was originally very wary to do something like this without notifying the user as it seems like it could baloon compilation times or otherwise be very odd if it happens silently by accident (e.g. we retry by accident too many times).

Could we perhaps only ignore emitting this if we're on CI or in UI testing mode? Because this happens so rarely I'd still want, by default, a user to see what happened and say "wait what that's not supposed to happen!" and then send a bug report to us.

Reply:
@alexcrichton The linker segfault retry won't be enabled unless RUSTC_RETRY_LINKER_ON_SEGFAULT=1, so this is already CI-specific.